src/plot.js

```diff
@@ -279,17 +327,57 @@ class Facet extends Mark {
       }
       marksChannels.push(markChannels);
     }
+    this.layout = function(index, scaleDescriptors, values, dimensions, options) {
```

Comment:
We shouldn’t assign this method as a side-effect of calling *facet*.initialize; it should be a method on the Facet class.